aiida/orm/nodes/process/calculation/calcjob.py

```diff
@@ -347,6 +352,7 @@ def set_scheduler_state(self, state: 'JobState') -> None:
         self.set_attribute(self.SCHEDULER_STATE_KEY, state.value)
         self.set_attribute(self.SCHEDULER_LAST_CHECK_TIME_KEY, timezone.datetime_to_isoformat(timezone.now()))
 
+    @field_method(f'attributes.{SCHEDULER_STATE_KEY}', key='scheduler_state', dtype=str)
     def get_scheduler_state(self) -> Optional['JobState']:
```

Comment:
Yeh, I like the idea of having the field “close” to the method, as it feels like it will make things like refactoring easier, but indeed the special case discrepancies are not ideal. I guess you will still have this problem if, as you mention, we want to look to auto generate the methods from the fields

Reply:
See https://github.com/aiidateam/aiida-core/pull/5088#issuecomment-994422493